tests/Tests.Shared/test-api.h

```diff
@@ -0,0 +1,599 @@
+                    try \
+                    { \
+                        ::testing::TestRunner::SetUp(test); \
+                    } \
+                    catch (const ::WEX::TestExecution::VerifyFailureException&) \
+                    { \
+                    } \
+                    \
+                    try \
+                    { \
+                        ::testing::TestRunner::TestBody(test); \
+                    } \
+                    catch (const ::WEX::TestExecution::VerifyFailureException&) \
+                    { \
+                    } \
+                    \
+                    try \
+                    { \
+                        ::testing::TestRunner::TearDown(test); \
+                    } \
+                    catch (const ::WEX::TestExecution::VerifyFailureException&) \
+                    { \
+                    } \
+                    _ExchangeAlternateTestName(std::string(), true); \
+                    \
+                    WEX::Logging::Log::EndGroup(currentTest.c_str()); \
+                    \
+                    delete factory; \
+                    delete test; \
+                } \
+            } \
+            WEX::Logging::Log::Comment(L"\nEnd of parameterized subtests for " L#className L"_" L#prefix L".\n"); \
+        } \
+    };
+
+// clang-format on
+
+__if_not_exists(GetCurrentTestDirectory) {
+    static std::string GetCurrentTestDirectory() {
+        WEX::Common::String testDeploymentPath;
+        WEX::TestExecution::RuntimeParameters::TryGetValue(L"TestDeploymentDir", testDeploymentPath);
+        std::wstring toReturn(testDeploymentPath);
+
+        static std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
+        return converter.to_bytes(toReturn);
+    }
+}
+
+__if_not_exists(GetTestFullName) {
+    static std::string GetTestFullName() {
+        std::string alternateTestName = _ExchangeAlternateTestName(std::string(), false);
+        if (!alternateTestName.empty()) {
+            return alternateTestName;
+        }
+        WEX::Common::String testName;
+        WEX::TestExecution::RuntimeParameters::TryGetValue(L"TestName", testName);
+
+        std::wstring wideToReturn(testName);
+        std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
+        std::string toReturn = converter.to_bytes(wideToReturn);
+
+        toReturn = toReturn.substr(0, toReturn.rfind("::Test"));
+        toReturn.replace(toReturn.find("_"), 1, ".");
+        return toReturn;
+    }
+}
+
+#ifndef LOG_TEST_PROPERTY
+#define LOG_TEST_PROPERTY(key, value)                                            \
+    {                                                                            \
+        static std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter; \
+        std::wstring wideKey = converter.from_bytes(key);                        \
+        std::wstring wideValue = converter.from_bytes(value);                    \
```

Comment:
Does Log::Property require`typeof(value) == std::wstring`? If not, and it's overloaded, we should pass value through unharmed for the appropriate implicit conversions to happen. This is too narrowly-specified as-is, it will only work for utf8 strings.